src/wandb_osh/lightning_hooks.py

```diff
@@ -49,4 +52,6 @@ def on_validation_epoch_end(
     ) -> None:
         if trainer.sanity_checking:
             return
-        self._hook()
+        # Call the hook every {frequency} epoch
+        if trainer.current_epoch % self.frequency == 0:
```

Comment:
I think this line should be implemented in `TriggerWandbSyncHook` (i.e., we pass the `frequency` argument there during `__init__` and then just call `self._hook()` here. This way, all derived classes can use this and not just the one for pytorch lightning (and then we add the same passing of the `frequency` argument to `TriggerWandbSyncRayHook`).

Reply:
That's better indeed, I will make the change